cpp/src/arrow/compute/exec/expression_benchmark.cc

```diff
@@ -120,5 +148,31 @@ BENCHMARK_CAPTURE(BindAndEvaluate, nested_array,
 BENCHMARK_CAPTURE(BindAndEvaluate, nested_scalar,
                   field_ref(FieldRef("struct_scalar", "float")));
 
+BENCHMARK_CAPTURE(ExecuteScalarExpressionOverhead, complex_expression, complex_expression)
+    ->ArgNames({"rows_per_batch"})
+    ->RangeMultiplier(10)
```

Comment:
Perhaps cut down on the number of concrete runs by passing 100 here?

Reply:
I think we can safely drop the 10M down to 1M.  I originally asked for 10M to see the effects when the batch size was greater than L3 but I don't think we are seriously considering doing things with that size of batch.  I think we can also adjust the lower limit to 1000.

Reply:
Changed the upper and lower limits of the range here, thanks!